### PR TITLE
1005 csv export

### PR DESCRIPTION
**Pull request by jasonmule**
This pull request makes the following changes:
- Adds an export button to export filtered posts. Currently, only CSV is supported.

Test these changes by:
- Navigating to the post list view and clicking export. Posts should be exported with the current filters.

Fixes ushahidi/platform#1005 .

Ping @ushahidi/platform

---
This change is [Reviewable](https://reviewable.io/reviews/ushahidi/platform-client/174)

### CODE REVIEW

File: server/www/templates/views/list.html [hunk 1]
```diff
@@ -18,9 +18,9 @@
                 </div>
             </div> -->
 
-            <!-- <button type="button" class="button-secondary alt icon-only-when-small export">
-                <span>Export</span>
-            </button> -->
+            <button type="button" class="button-secondary alt icon-only-when-small export" ng-click="exportPosts()">
+                <span translate>post.post_actions.export</span>
+            </button>
```
Comment (jasonmule) [inline server/www/templates/views/list.html]: The button needs to be available to non-admin users as well.

**Comment by jasonmule** [comment]
@willdoran @rjmackay Could you review?

File: server/www/templates/views/list.html (hunk 1 above)
Comment (willdoran) [inline server/www/templates/views/list.html]: Should this check for the Bulk data permissions?

Comment (jasonmule) [inline server/www/templates/views/list.html]: Probably not. For now it's positioned with the other bulk data actions but I should disable the check.

File: app/post/directives/views/post-view-list-directive.js [hunk 2]
```diff
@@ -110,6 +112,47 @@ function (
...
+                                href: 'data:attachment/' + format + ';charset=utf-8,' + encodeURIComponent(response[1].data),
+                                target: '_self',
+                                download: siteName + '.csv'
+                            });
+
+                            // Save to file
+                            anchor[0].click();
+
+                            // ... and finally remove the link
+                            anchor.remove();
```
Comment (willdoran) [inline app/post/directives/views/post-view-list-directive.js]: This is quite of neat.

**Comment by willdoran** [comment]
This looks great.

**Comment by rjmackay** [comment]
Reviewed 2 of 5 files at r1, 1 of 1 files at r2, 2 of 2 files at r3.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/ushahidi/platform-client/174)*

**Pull request merged**